### ISSUE TITLE
Cannot start using NPM

### ISSUE THREAD

**Issue by GrantBaum** [MyMICDS/MyMICDS-Admin-Panel#1]
The server listens but fails to compile the angular and a critical error occurs.

**Comment by michaelgira23** [comment]
I believe the Admin Panel doesn't actually have any meaningful code for it, and it was never pushed to production. If we want to actually create an admin panel, we should probably just purge the existing files and start a fresh new Angular project.

**Comment by GrantBaum** [comment]
why not make it in TypeScript?

**Comment by michaelgira23** [comment]
The existing code is already in TypeScript, and it also uses the Angular framework (written in the TypeScript programming language). We chose to create the Admin Panel in Angular just to remain consistent with the existing MyMICDS Angular frontend.

**Comment by GrantBaum** [comment]
Ok. What if I have no clue how to work angular? I don't see why the consistency should be prioritized.

**Comment by michaelgira23** [comment]
That is actually a very good point, and it's something we've talked about before. Ultimately, we wrote MyMICDS in Angular because it was a popular web framework at the time and we thought it would help us create a better-organized frontend. It's possible, but it would be a large undertaking to re-write the entirety of MyMICDS again, so the main frontend will remain in Angular for the foreseeable future.

Practically, if the Admin Panel is also in Angular, then we can copy-paste some components already used in the main frontend like maybe authentication logic or other UI. Also, if we use a simpler web framework, then someone working on the admin panel has to two learn two frameworks instead of one, which we think is a larger barrier of entry than before. I suppose one assumption is that a developer would first work on the public frontend before working on the admin panel which is only for the developers.

One benefit to using Angular though is that it's a pretty advanced framework, so if you manage to learn it, then you can transfer that knowledge to other projects, and knowing Angular is great for your resume 😄 There are a lot of resources out there to learn Angular (such as [Angular University](https://angular-university.io), which is what I used). I'm also happy to talk in the Slack if you have more in-depth questions about Angular.

The above is the reasoning is why we wrote the Admin Panel in Angular, but nothing is set in stone. Because I don't think anyone else is actively working on this project, feel free to do whatever you deem best!

**Comment by GrantBaum** [comment]
ok! do I need all the stuff that was generated here for an angular project to work, or can I just start fresh?

**Comment by michaelgira23** [comment]
No, we could completely delete this repo’s existing contents because I don’t think we’ve significantly modified it from the template project generated by [Angular CLI](https://angular.io/cli)

**Comment by GrantBaum** [comment]
I made the typescript project, if there is anything I need to add LMK
